l10n_it_fatturapa_in/models/account.py

```diff
@@ -158,9 +158,11 @@ def remove_attachment_link(self):
         return {'type': 'ir.actions.client', 'tag': 'reload'}
 
     @api.model
-    def compute_xml_amount_untaxed(self, DatiRiepilogo):
-        amount_untaxed = 0.0
-        for Riepilogo in DatiRiepilogo:
+    def compute_xml_amount_untaxed(self, FatturaBody):
```

Comment:
Se possibile non cambierei la firma di `compute_xml_amount_untaxed` , perchè se ci fossero altri moduli (penso alle personalizzazioni) che ne fanno override, si romperebbero

Reply:
@eLBati Sono d'accordo con te, ho dovuto farlo perchè come puoi vedere nel codice non potremmo raggiungere ``FatturaElettronicaBody.DatiGenerali.DatiGeneraliDocumento.Arrotondamento`` obbligandoci a splittare il calcolo sul chiamante del metodo

Reply:
Concordo che cambiare la firma del metodo è male.
Per evitarlo vedo due strade:
* Aggiungi FatturaBody come `kwarg`, credo che in questo modo i metodi che fanno override non dovrebbero avere ripercussioni;
* (consigliata) Aggiungi il valore dell'arrotondamento nel `context` di `self`.

Consiglio la seconda perché secondo me l'arrotondamento è un valore che può influire su diversi campi della fattura, ma non per questo va passato in ogni metodo, quindi il `context` mi sembra il posto migliore.

Inoltre, se un metodo si chiama `compute_xml_amount_untaxed`, credo abbia senso che il suo argomento sia `DatiRiepilogo`, dove si trova il valore che vogliamo recuperare.

Reply:
Penso che se il metodo calcola l'amount untaxed, questo dovrebbe avere tutto il necessario per poter fare ciò a prescindere da dove fisicamente si trovano i dati nell'XML, quindi il metodo nasconde i dettagli del calcolo a prescindere da come siano essi prelevati alla fonte. 

L'unica reale soluzione che risolvebbe il problema è passare il FatturaBody perchè li dentro abbiamo tutto il necessario per fare i calcoli di dovere.

Tuttavia, se il problema è effettivamente che ci sono dei moduli che usano questo metodo, questi saranno sicuramente molto pochi, quindi l'impatto è minimo. Volendo possiamo anche aggiungere un check sul tipo di dato che viene passato e fare una raise per avvisare l'utente del problema ... questo avviserà la società di dovere. In qualunque caso il fix risulterebbe piuttosto banale per chi ha customizzato il modulo.

Passarlo come ``kwargs`` o nel ``context`` sarebbe giusto una soluzione che comunque non restituirebbe il valore corretto dell'untaxed amount dato che chi ridefinisce il metodo comunque non passarebbe questi valori ne come ``kwargs`` e neanche nel ``context``. Pertanto per restituire il valore giusto bisognerebbe mettere mano al codice, ma a quel punto tanto vale fare la cosa giusta no?

Se si vuole avere l'implementazione corretta, a mio parere il codice dovrebbe rimanere così com'è adesso ed i chiamanti del metodo, o chi lo ridefinisce, dovrebbero essere modificati/adattati se vuole ottenere i valori corretti.

Reply:
@robyf70 allora per me ok tenerlo così. Puoi descrivere il problema nel file `ROADMAP.rst`?

Reply:
@eLBati Ok